app/utils/DateErrorFormatter.scala

```diff
@@ -16,12 +16,25 @@
 
 package utils
 
+import play.api.data.FormError
 import play.api.i18n.Messages
 
 object DateErrorFormatter {
 
   def formatArgs(args: Seq[Any])(implicit messages: Messages): Seq[String] = {
-    args.map(arg => messages(s"date.$arg").toLowerCase)
+    val dateArgs = Seq("day", "month", "year")
+    args.map(arg => if (dateArgs.contains(arg)) messages(s"date.$arg").toLowerCase else arg.toString)
   }
 
+  def addErrorClass(error: Option[FormError], dateArg: String): String = {
```

Comment:
This can be replaced with collect / getOrElse

    error.collect{case fe if fe.args.isEmpty || fe.args.contains(dateArg) => "govuk-input--error"}.getOrElse("")

Reply:
I like that change, so I've applied it